audio/out/buffer.c

```diff
@@ -698,12 +698,12 @@ static MP_THREAD_VOID playthread(void *arg)
 
         // Wait until the device wants us to write more data to it.
         // Fallback to guessing.
-        double timeout = INFINITY;
+        int64_t timeout = INT64_MAX;
         if (p->streaming && !retry && (!p->paused || ao->stream_silence)) {
             // Wake up again if half of the audio buffer has been played.
             // Since audio could play at a faster or slower pace, wake up twice
             // as often as ideally needed.
-            timeout = ao->device_buffer / (double)ao->samplerate * 0.25;
+            timeout = MP_TIME_S_TO_NS(ao->device_buffer / (double)ao->samplerate * 0.25);
```

Comment:
The comment says "half", but the factor is 0.25

Reply:
But then it is clarified
> Since audio could play at a faster or slower pace, wake up twice as often as ideally needed.
Maybe not best comment, but I don't want to change it.

Reply:
Hmm yeah I interpreted that differently at first, but I guess it's fine.

Reply:
https://github.com/mpv-player/mpv/commit/baa7b5c8dd35365a8a62aa39c84394e5315b79e0

Reply:
Yeah I already looked at that while you answered before. The initial expectation was to find a commit that changed 0.5 to 0.25 but didn't update the comment, but turns out I misunderstood it.